Sources/AblyAssetTrackingInternal/Utils/Codable+Extensions.swift

```diff
@@ -1,4 +1,22 @@
 import Foundation
+import AblyAssetTrackingCore
+
+public extension Encodable {
+    /**
+     Utility function to construct JSON string from given `Encodable` object.
+     - Throws:
+        - ErrorInformation of JSONCodingError type when we are not able to create `String` from coded `Data`
+        - Any JSONDecoder error: Thrown by the `JSONEncoder` and just passed forward
+     - Returns: JSON string encoded with UTF-8
+     */
+    func toJSONString() throws -> String {
+        let data = try JSONEncoder().encode(self)
+        if let result =  String(data: data, encoding: .utf8) {
+            return result
+        }
+        throw ErrorInformation(type: .JSONCodingError(for: "\(data.self)"))
```

Comment:
in this case, the reason for this exception is not clear. It should inform the user that 
```swift
String(data: data, encoding: .utf8)
```
can't create String from data

Reply:
This is also code I did not write, but just moved because I needed to move them into a new module.

Reply:
😅